purchase_request/models/purchase_request.py

```diff
@@ -0,0 +1,251 @@
+# -*- coding: utf-8 -*-
+# Copyright 2016 Eficent Business and IT Consulting Services S.L.
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl-3.0).
+
+from openerp import api, fields, models
+import openerp.addons.decimal_precision as dp
+
+_STATES = [
+    ('draft', 'Draft'),
+    ('to_approve', 'To be approved'),
+    ('approved', 'Approved'),
+    ('rejected', 'Rejected')
+]
+
+
+class PurchaseRequest(models.Model):
+
+    _name = 'purchase.request'
+    _description = 'Purchase Request'
+    _inherit = ['mail.thread', 'ir.needaction_mixin']
+
+    @api.model
+    def _company_get(self):
+        company_id = self.env['res.company']._company_default_get(self._name)
+        return self.env['res.company'].browse(company_id.id)
+
+    @api.model
+    def _get_default_requested_by(self):
+        return self.env['res.users'].browse(self.env.uid)
+
+    @api.model
+    def _get_default_name(self):
+        return self.env['ir.sequence'].get('purchase.request')
+
+    @api.model
+    def _default_picking_type(self):
+        type_obj = self.env['stock.picking.type']
+        company_id = self.env.context.get('company_id') or \
+            self.env.user.company_id.id
+        types = type_obj.search([('code', '=', 'incoming'),
+                                 ('warehouse_id.company_id', '=', company_id)])
+        if not types:
+            types = type_obj.search([('code', '=', 'incoming'),
+                                     ('warehouse_id', '=', False)])
+        return types[:1]
```

Comment:
Is it not better to add a boolean field on stock.picking.type to set the default picking type for purchase request (With a unique constraint) ?